brillig_vm/src/registers.rs

```diff
@@ -1,42 +1,23 @@
 use crate::Value;
 use serde::{Deserialize, Serialize};
 
-/// Registers will store field element values during the
-/// duration of the execution of the bytecode.
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub struct Registers {
-    pub inner: Vec<Value>,
-}
-
-impl IntoIterator for Registers {
-    type Item = Value;
-    type IntoIter = RegistersIntoIterator;
-
-    fn into_iter(self) -> Self::IntoIter {
-        RegistersIntoIterator { registers: self, index: 0 }
-    }
-}
-pub struct RegistersIntoIterator {
-    registers: Registers,
-    index: usize,
+    // Keep track of registers as an option to represent uninitialized registers
+    // These occur when we set a register value at a noncontiguous index, past currently defined registers
+    // This could just store 0's, but it would be
+    pub inner: Vec<Option<Value>>,
```

Comment:
Unfinished comment? Also this is still a bit confusing to me.... Can't follow why you'd need/want this.